variables/templates_test.go

```diff
@@ -0,0 +1,168 @@
+	type test struct {
+		vc           VariableConfig
+		path         string
+		wantErr      error
+		wantContents string
+	}
+
+	tests := []test{
+		{
+			vc:           VariableConfig{setVariableMap: SetVariableMap{}, applicationTemplates: map[string]*TextTemplate{}},
+			path:         "non-existent.test",
+			wantErr:      errors.New("open non-existent.test: no such file or directory"),
+			wantContents: start,
+		},
+		{
+			vc: VariableConfig{
+				templatePrefix: "PREFIX",
+				setVariableMap: SetVariableMap{
+					"REPLACE_ME": {Value: "VAR_REPLACED"},
+				},
+				constants: []Constant{{Name: "REPLACE_ME", Value: "CONST_REPLACED"}},
+				applicationTemplates: map[string]*TextTemplate{
+					"###PREFIX_APP_REPLACE_ME###": {Value: "APP_REPLACED"},
+				},
+			},
+			path:         testPath,
+			wantErr:      nil,
+			wantContents: simple,
+		},
+		{
+			vc: VariableConfig{
+				templatePrefix: "PREFIX",
+				setVariableMap: SetVariableMap{
+					"REPLACE_ME": {Value: "VAR_REPLACED\nVAR_SECOND"},
+				},
+				constants: []Constant{{Name: "REPLACE_ME", Value: "CONST_REPLACED\nCONST_SECOND"}},
+				applicationTemplates: map[string]*TextTemplate{
+					"###PREFIX_APP_REPLACE_ME###": {Value: "APP_REPLACED\nAPP_SECOND"},
+				},
+			},
+			path:         testPath,
+			wantErr:      nil,
+			wantContents: multiline,
+		},
+		{
+			vc: VariableConfig{
+				templatePrefix: "PREFIX",
+				setVariableMap: SetVariableMap{
+					"REPLACE_ME": {Value: "VAR_REPLACED\nVAR_SECOND", Variable: Variable{AutoIndent: true}},
+				},
+				constants: []Constant{{Name: "REPLACE_ME", Value: "CONST_REPLACED\nCONST_SECOND", AutoIndent: true}},
+				applicationTemplates: map[string]*TextTemplate{
+					"###PREFIX_APP_REPLACE_ME###": {Value: "APP_REPLACED\nAPP_SECOND", AutoIndent: true},
+				},
+			},
+			path:         testPath,
+			wantErr:      nil,
+			wantContents: autoIndent,
+		},
+		{
+			vc: VariableConfig{
+				templatePrefix: "PREFIX",
+				setVariableMap: SetVariableMap{
+					"REPLACE_ME": {Value: "file.test", Variable: Variable{Type: FileVariableType}},
+				},
+				constants: []Constant{{Name: "REPLACE_ME", Value: "CONSTs Don't Support File"}},
+				applicationTemplates: map[string]*TextTemplate{
+					"###PREFIX_APP_REPLACE_ME###": {Value: "file.test", Type: FileVariableType},
+				},
+			},
+			path:         testPath,
+			wantErr:      nil,
+			wantContents: file,
+		},
+	}
+
+	for _, tc := range tests {
+		setTestPathContents()
+
+		gotErr := tc.vc.ReplaceTextTemplate(tc.path)
+		if gotErr != nil && tc.wantErr != nil {
+			if gotErr.Error() != tc.wantErr.Error() {
+				t.Fatalf("wanted err: %s, got err: %s", tc.wantErr, gotErr)
+			}
+		} else if gotErr != nil {
+			t.Fatalf("got unexpected err: %s", gotErr)
+		} else {
+			gotContents, _ := os.ReadFile(tc.path)
+			if string(gotContents) != tc.wantContents {
+				t.Fatalf("wanted contents: %s, got contents: %s", tc.wantContents, string(gotContents))
+			}
+		}
+
+		cleanTestPath()
```

Comment:
This should be deferred so it gets deleted on failure

Reply:
or run in a `t.Cleanup` (I would prefer the file get placed in a `t.TempDir` anyways)